MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -1425,23 +1426,26 @@ public void addPart(Part p, int transitDays) {
                         ((Armor) spare).setAmount(((Armor) spare).getAmount()
                                 + ((Armor) p).getAmount());
                         MekHQ.triggerEvent(new PartChangedEvent(spare));
+                        p.setId(-1);
```

Comment:
What does the -1 part ID signify?

Reply:
Code which writes parts in some areas uses a full part description if the ID is less than or equal to zero. This is an attempt to force that if that code is the cause of the refit bugs.

Reply:
(The full description can't have an ID get out of sync, because it creates a brand new part)